### ISSUE TITLE
Unknown reason cause Pub.dev Null safety check failed

### ISSUE THREAD

**Issue by hythl0day** [dart-lang/pana#910]
I uploaded a package on Pub.dev, [(hetu_script)](https://pub.dev/packages/hetu_script/score) and there's an issue in the null safety check:

![image](https://user-images.githubusercontent.com/2274141/122118816-12577780-ce5b-11eb-921f-f5ddbfca15d2.png)

I checked all import statements in my project and run pana locally multiple times, and I could not find out why there's this issue.
(the previous version of my package has passed the analysis)

the local pana result are:

```
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe --version`...
INFO       Running `flutter.bat --no-version-check --version --machine`...
WARNING    pana might update or modify files in `.`.
           Analysis will begin in 15 seconds, hit CTRL+C to abort it.
           To remove this message, use `--no-warning`.
INFO       Running `git rev-parse --show-toplevel`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe pub upgrade --verbose`...
INFO       Analyzing package...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe analyze --format machine bin`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe analyze --format machine lib`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dartfmt.bat --dry-run --set-exit-if-changed C:\Users\aleph42\AppData\Local\Temp\pana_33a166f9\.\bin`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dartfmt.bat --dry-run --set-exit-if-changed C:\Users\aleph42\AppData\Local\Temp\pana_33a166f9\.\lib`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe pub get`...
INFO       Running `D:\SDK\Dart\dart-sdk\bin\dart.exe pub outdated --json --up-to-date --no-dev-dependencies --no-dependency-overrides`...

## ✓ Follow Dart file conventions (20 / 20)
### [*] 10/10 points: Provide a valid `pubspec.yaml`


### [*] 5/5 points: Provide a valid `README.md`


### [*] 5/5 points: Provide a valid `CHANGELOG.md`


## ✓ Provide documentation (10 / 10)
### [*] 10/10 points: Package has an example

* Found example at: `example/example.dart`

## ✗ Support multiple platforms (10 / 20)
### [~] 10/20 points: Supports 1 of 2 possible platforms (**native**, js)

Consider supporting multiple platforms:

<details>
<summary>
Package not compatible with runtime js
</summary>

Because:
* `package:hetu_script/hetu_script.dart` that imports:
* `package:hetu_script/source/source_provider.dart` that imports:
* `dart:io`
</details>

## ✓ Pass static analysis (30 / 30)
### [*] 30/30 points: code has no errors, warnings, lints, or formatting issues


## ✓ Support up-to-date dependencies (20 / 20)
### [*] 10/10 points: All of the package dependencies are supported in the latest version

|Package|Constraint|Compatible|Latest|
|:-|:-|:-|:-|
|[`args`]|`^2.1.0`|2.1.1|2.1.1|
|[`characters`]|`^1.1.0`|1.1.0|1.1.0|
|[`meta`]|`^1.4.0`|1.4.0|1.4.0|
|[`path`]|`^1.8.0`|1.8.0|1.8.0|
|[`pub_semver`]|`^2.0.0`|2.0.0|2.0.0|
|[`quiver`]|`^3.0.1`|3.0.1|3.0.1|

<details><summary>Transitive dependencies</summary>

|Package|Constraint|Compatible|Latest|
|:-|:-|:-|:-|
|[`collection`]|-|1.15.0|1.15.0|
|[`matcher`]|-|0.12.10|0.12.10|
|[`stack_trace`]|-|1.10.0|1.10.0|
</details>

To reproduce run `dart pub outdated --no-dev-dependencies --up-to-date --no-dependency-overrides`.

[`args`]: https://pub.dev/packages/args
[`characters`]: https://pub.dev/packages/characters
[`meta`]: https://pub.dev/packages/meta
[`path`]: https://pub.dev/packages/path
[`pub_semver`]: https://pub.dev/packages/pub_semver
[`quiver`]: https://pub.dev/packages/quiver
[`collection`]: https://pub.dev/packages/collection
[`matcher`]: https://pub.dev/packages/matcher
[`stack_trace`]: https://pub.dev/packages/stack_trace


### [*] 10/10 points: Package supports latest stable Dart and Flutter SDKs


## ✓ Support sound null safety (20 / 20)
### [*] 20/20 points: Package and dependencies are fully migrated to null safety!


Points: 110/120.
```

And the issue appears repeatedly after I uploads.

**Comment by isoos** [comment]
This may be a bug in `package:analyzer`, as I can't reproduce this locally with the latest HEAD. We are planning to release this soon on pub.dev, please be patient, it will re-analyze the package automatically and fix the analysis results too.

**Comment by hythl0day** [comment]
> This may be a bug in `package:analyzer`, as I can't reproduce this locally with the latest HEAD. We are planning to release this soon on pub.dev, please be patient, it will re-analyze the package automatically and fix the analysis results too.

I just found out that I used a reserved keyword 'factory' as a enum item in that file. I guess this might be the reason. I didn't notice this previously because the VSCODE didn't report any error about this. And my highlight of text didn't show the different. When I open my project on my home computer which use a slightly different highlight I notice the keyword. I tried re-upload the  package. And I'm waiting for this time's analysis result.
